Small refactor in datapath/linux/node.go

derailed opened the pull request:
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [x] Thanks for contributing!

This PR addresses a few small clean up items in datapath/linux/node.go

- Fix typos
- Dry up valid IP check
- Ensure errors are propagated

Fixes: #issue-number

Signed-off-by: Fernand Galiana <fernand.galiana@isovalent.com>

gentoo-root reviewed (commented):
The changes mentioned in the commit message seem no longer here:

> - Fix typo
> - Dry up valid IP check
> - Ensure errors are propagated

gentoo-root commented on pkg/datapath/linux/ipsec.go:
```diff
@@ -98,8 +98,10 @@ func (n *linuxNodeHandler) enableIPsec(newNode *nodeTypes.Node, nodeID uint16) e
 func (n *linuxNodeHandler) enableIPsecIPv4(newNode *nodeTypes.Node, nodeID uint16, zeroMark bool) error {
-	var spi uint8
-	var errs error
+	var (
```
Does the syntax with a single var block have any benefits?

IMO, unrelated styling changes should be avoided if both styles are permitted, because:

1. It makes `git blame` less useful.
2. It could make backporting and reverting more troublesome.
3. It adds some noise that interferes with reviewing.

derailed replied:
No benefits besides DRY, line of sight and declaring vars where they are used.
I'll remove it.

pchaigno reviewed (changes requested):
Thanks for the pull request!

It seems that it fixes a case where an error isn't reported. That's a fix and should be in its own pull request that we can separately backport.

In addition, it would be good to split the changes in multiple commits with proper descriptions as this is going to cause rebase & backport conflicts for other works; having a proper commit history helps resolution of those conflicts.

derailed committed:
Dry up ip validation check

Validate Ipv4/Ipv6 was replicated across several calls.
Introduce helper to perform the checks to a single location.

Signed-off-by: Fernand Galiana <fernand.galiana@isovalent.com>

derailed committed:
Fix typo in datapath/linux/ipsec.go

previousa -> previous

Signed-off-by: Fernand Galiana <fernand.galiana@isovalent.com>

derailed committed:
Refactor: Pre allocate slices where applicable

To avoid unnecessary GC churn ensure slices are pre allocated.

Signed-off-by: Fernand Galiana <fernand.galiana@isovalent.com>

pchaigno commented on pkg/datapath/linux/node.go:
```diff
@@ -1213,8 +1213,7 @@ func (n *linuxNodeHandler) NodeConfigurationChanged(newConfig datapath.LocalNode
 
-			var targetDevices []string
-			targetDevices = append(targetDevices, option.Config.DirectRoutingDevice)
+			targetDevices := []string{option.Config.DirectRoutingDevice}
```
Any reason not to use `make` with the length of `option.Config.GetDevices()` + 1 here? (Note I'm a Go newbie.)

derailed replied:
Good catch!

pchaigno commented on pkg/datapath/linux/node.go:
```diff
@@ -1401,8 +1399,10 @@ func (n *linuxNodeHandler) NodeCleanNeighborsLink(l netlink.Link, migrateOnly bo
+	var (
+		neighSucceeded, neighErrored int
+		which                        string
+	)
```
I'd avoid making changes like this and the one on line 1287. They are style preferences for which the Cilium project doesn't have rules AFAIK. They don't fix anything but can cause conflicts on backports & co. IMO, the drawbacks outweigh the benefits.

derailed replied:
I agree besides DRY and readability no value add here. I'll revert it

pchaigno reviewed (approved):
Thanks for reworking the commit history!
